### DISCUSSION TITLE
Issue with the AKLT chain in quspin

### DISCUSSION THREAD

**Discussion by mahmudams** [QuSpin/QuSpin#662] [Q&A]
I am trying to implement the AKLT model using Quspin. The code I have written is below:
<img width="990" alt="Screenshot 2024-02-08 at 6 33 08 AM" src="https://github.com/QuSpin/QuSpin/assets/139525936/f00d621c-d5e8-4756-a784-dc01a2c84915">

Setting S =1 initializes the spin-1 basis. But when I am calling the spin operators Sx, Sy, and Sz defined in the code, it is not working. Can someone please point out how to fix this issue?

**Comment by mgbukov** [comment]
there are only the '+', '-', and 'z' operators for higher spin; you'd have
to express the operators in terms of these and it'll work.

On Thu, Feb 8, 2024 at 1:36 PM mahmudams ***@***.***> wrote:

> I am trying to implement the AKLT model using Quspin. The code I have
> written is below:
> Screenshot.2024-02-08.at.6.33.08.AM.png (view on web)
> <https://github.com/QuSpin/QuSpin/assets/139525936/f00d621c-d5e8-4756-a784-dc01a2c84915>
>
> Setting S =1 initializes the spin-1 basis. But when I am calling the spin
> operators Sx, Sy, and Sz defined in the code, it is not working. Can
> someone please point out how to fix this issue?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/QuSpin/QuSpin/discussions/662>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADHSUBVBGUBVZQUOASJS4GDYSTBEZAVCNFSM6AAAAABC7VDEYSVHI2DSMVQWIX3LMV43ERDJONRXK43TNFXW4OZWGE4TKNBWGE>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>


-- 
Marin Bukov, PhD
Research Group Leader
Max Planck Institute for the Physics of Complex Systems
Noethnitzer Str. 38
Dresden 01187, Germany

url: https://www.pks.mpg.de/nqd
<http://quantum-dynamics.phys.uni-sofia.bg/>

**Reply by mahmudams** [reply]
Expressing the operators in terms '+', '-', and 'z,' now things are working. Thank you!
